packages/ra-core/src/form/useFormGroup.ts

```diff
@@ -63,16 +66,27 @@ type FormGroupState = {
  * @returns {FormGroupState} The form group state
  */
 export const useFormGroup = (name: string): FormGroupState => {
-    const { dirtyFields, touchedFields, errors } = useFormState();
+    const { dirtyFields, touchedFields, validatingFields, errors } =
+        useFormState();
+
+    // dirtyFields, touchedFields and validatingFields are objects with keys being the field names
+    // Ex: { title: true }
+    // However, they are not correctly serialized when using JSON.stringify
+    // To avoid our effects to not be triggered when they should, we extract the keys and use that as a dependency
+    const dirtyFieldsNames = Object.keys(dirtyFields);
+    const touchedFieldsNames = Object.keys(touchedFields);
+    const validatingFieldsNames = Object.keys(validatingFields);
```

Comment:
will this work with nested inputs? :thinking:

Reply:
Yes, react-hook-form field name for deep fields are `one.two.three`